### ISSUE TITLE
Gradient Clipping for FSDP

### ISSUE THREAD

**Issue by yunjae-won** [ContextualAI/HALOs#15]
Hi! Thank you for maintaining such a valuable repository.
I would like to suggest a minor fix regarding gradient clipping. For FSDP, we should not use  `torch.nn.utils.clip_grad_norm_` ([relevant issue]( https://github.com/pytorch/pytorch/issues/72548#issue-1127816211)), but instead directly call the `clip_grad_norm_` method of  the FSDP module. Thus, I would like to suggest modifying the following:

https://github.com/ContextualAI/HALOs/blob/f9f78263cb7aa70809a87f9f74482dfd09f8c60b/trainers.py#L453-L455

into the following:
```python
        def clip_gradient(self):
            """Clip the gradient norm of the parameters."""
            if self.fsdp:
                return self.policy.clip_grad_norm_(self.config.model.max_grad_norm).item()
            return torch.nn.utils.clip_grad_norm_(self.policy.parameters(), self.config.model.max_grad_norm).item()
```
Thank you so much!

**Comment by kawine** [comment]
fixed! thanks for pointing this out @YJWon99 !
